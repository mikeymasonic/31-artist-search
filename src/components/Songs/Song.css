@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300&display=swap');

.song{
  height: 3vw;
  width: 12vw;
  display: flex;
  justify-content: center;
  text-align: center;
  margin: 0 3vw 3vw 0;
  padding: 1vw;
  border: .2vw solid rgb(133, 220, 226);
  border-radius: 1vw;
  font-size: 1.1vw;
  font-family: 'Montserrat', sans-serif;
  font-weight: 200;
}

.songlink {
  text-decoration: none;
  color: #f7d89e;
}

.songlink:hover {
  color: white;
}

.song:hover {
  background: hotpink;
}
