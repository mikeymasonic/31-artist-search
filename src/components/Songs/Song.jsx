import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Song.css';

const Song = ({ songTitle, artistName }) => {
  return (
    <Link className={styles.songlink} to={`/lyrics/${artistName}/${songTitle}`}>
      <h3 className={styles.song}>{songTitle}</h3>
    </Link>
  );
};

Song.propTypes = {
  songTitle: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired
};

export default Song;
