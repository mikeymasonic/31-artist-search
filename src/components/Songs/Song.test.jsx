import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song';

describe('Song component', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song artistName='Silkworm' songTitle="Never Met A Man I didn\'t Like" />);
    expect(wrapper).toMatchSnapshot();
  });
});
