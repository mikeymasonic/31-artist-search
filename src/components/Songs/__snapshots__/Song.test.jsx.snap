// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component renders Song 1`] = `
<Link
  className="songlink"
  to="/lyrics/Silkworm/Never Met A Man I didn\\\\'t Like"
>
  <h3
    className="song"
  >
    Never Met A Man I didn\\'t Like
  </h3>
</Link>
`;
